File: 2050-count-good-numbers/2050-count-good-numbers.java
class Solution {
    int mod = (int)1e9+7;
    private long findPower(long a, long b){
        if(b == 0)
            return 1;
        long half = findPower(a, b/2);
        long ans = (half * half) % mod;
        if(b%2 == 1){
            ans = (ans*a)%mod;
        }
        return ans;
    }
    public int countGoodNumbers(long n) {
        long evenInd = (n+1)/2;
        long oddIndx = n/2;
        long ans = (findPower(5,evenInd) * findPower(4 , oddIndx)) % mod;
        return (int)ans;
    }
}

File: 2050-count-good-numbers/2050-count-good-numbers.cpp
class Solution {
public:
    int mod = 1e9 + 7;
    int findPower(long long a, long long b){
        if(b == 0)
            return 1;
        long long half = findPower(a, b/2);
        long long ans = (half * half)%mod;

        if(b % 2 == 1){
            ans = (ans*a)%mod;
        }
        return ans;
    }
    int countGoodNumbers(long long n) {
        long long ans = (long long)findPower(5, (n+1)/2) * findPower(4 , n/2) % mod;
        return ans;
    }
};